src/App.tsx

```diff
@@ -1,26 +1,205 @@
-/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import * as todosService from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { TodoItem } from './components/TodoItem/TodoItem';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMesage';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [isLoadingError, setIsLoadingError] = useState('');
+  const [isAddError, setIsAddError] = useState('');
+  const [isUpdateError, setIsUpdateError] = useState('');
+  const [isDeleteError, setIsDeleteError] = useState('');
+
+  const [status, setStatus] = useState('all');
+  const [todo, setTodo] = useState<Todo>();
+  const [todosIsLoading, setTodosIsLoading] = useState<number[]>([]);
+  const [isInputDisabled, setInputDisabled] = useState(false);
+  const [isFocus, setIsFocus] = useState(true);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setIsLoadingError('Unable to load todos');
+      })
+      .finally(() => {});
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    let fltrdTodos: Todo[] | undefined = todos;
+
+    switch (status) {
+      case 'all':
+        fltrdTodos = todos;
+        break;
+      case 'active':
+        fltrdTodos = todos?.filter(td => td.completed === false);
+        break;
+      case 'completed':
+        fltrdTodos = todos?.filter(td => td.completed === true);
+        break;
+    }
+
+    return fltrdTodos;
+  }, [status, todos]);
+
+  const handleClick = (event: React.MouseEvent) => {
+    setStatus(event.currentTarget.innerHTML.toLowerCase());
+  };
+
+  function addTodo({ userId, title, completed }: Todo) {
```

Comment:
It is important to stick to one way of creating functions everywhere
```suggestion
  const addTodo = ({ userId, title, completed }: Todo) => {
```